### PR TITLE
xpay maxfee fix

### PR DESCRIPTION
**Pull request by rustyrussell**
Fixes: #7888

### CODE REVIEW

File: plugins/askrene/flow.c [hunk 1]
```diff
@@ -176,6 +176,18 @@ const char *fmt_flows_step_scid(const tal_t *ctx,
 	return fmt_short_channel_id_dir(ctx, &scidd);
 }
 
+const char *fmt_flow(const tal_t *ctx,
```
Comment (niftynei) [inline plugins/askrene/flow.c:179]: oh this is nice

**Review by niftynei — APPROVED** [review]
ACK d13f977e4a8e1d9ceba9f088013b2f1da37705e0

**Comment by niftynei** [comment]
the Liquid testnet is having some trouble 

```
lightningd-1 2024-11-30T08:27:30.932Z DEBUG   plugin-cln-askrene: notify msg broken: We had 1 flows (0:9887833msat via 103x1x0/1->0x3301x1646/0->0x90x2024/1->63x90x29364/1->16x63x18351/1->16x21x18180/0->21x59x20149/1), but none have extra capacity and we need to send 12167msat more for fees. Couldn't find an htlc_max limited flow.
lightningd-1 2024-11-30T08:27:30.932Z **BROKEN** plugin-cln-askrene: \"-c:xpay#28/cln:xpay#157/cln-xpay:getroutes#40/cln:getroutes#240\": We had 1 flows (0:9887833msat via 103x1x0/1->0x3301x1646/0->0x90x2024/1->63x90x29364/1->16x63x18351/1->16x21x18180/0->21x59x20149/1), but none have extra capacity and we need to send 12167msat more for fees. Couldn't find an htlc_max limited flow.
```

**Comment by rustyrussell** [comment]
> the Liquid testnet is having some trouble
> 
> ```
> lightningd-1 2024-11-30T08:27:30.932Z DEBUG   plugin-cln-askrene: notify msg broken: We had 1 flows (0:9887833msat via 103x1x0/1->0x3301x1646/0->0x90x2024/1->63x90x29364/1->16x63x18351/1->16x21x18180/0->21x59x20149/1), but none have extra capacity and we need to send 12167msat more for fees. Couldn't find an htlc_max limited flow.
> lightningd-1 2024-11-30T08:27:30.932Z **BROKEN** plugin-cln-askrene: \"-c:xpay#28/cln:xpay#157/cln-xpay:getroutes#40/cln:getroutes#240\": We had 1 flows (0:9887833msat via 103x1x0/1->0x3301x1646/0->0x90x2024/1->63x90x29364/1->16x63x18351/1->16x21x18180/0->21x59x20149/1), but none have extra capacity and we need to send 12167msat more for fees. Couldn't find an htlc_max limited flow.
> ```

Yes, and it's fascinating why... I've reproduced here and it wasn't what I thought. Good to exercise this code path though....

**Pull request merged**